### ISSUE TITLE
not able to mount applications in sub directories, but only sometimes

### ISSUE THREAD

**Issue by plockaby** [unbit/uwsgi#263]
This is related to this ticket, sort of: https://github.com/unbit/uwsgi/issues/242

Here's my PSGI file:

    #!/usr/bin/env perl
    
    use strict;
    use warnings;
    use Data::Dumper;
    
    return sub {
        my $env = shift;
        return [
            '200',
            [ 'Content-Type' => 'text/html' ],
            [ "<h1>Hello World</h1><pre>" . Dumper($env) . "</pre>" ],
        ];
    };

Here is my INI file:

    [uwsgi]
    chdir = /data/apps/test
    psgi = index.psgi
    http-socket = :3001
    http-socket-modifier1 = 5
    scgi-socket = :4001
    scgi-modifier1 = 5
    master = true
    threads = 20
    logto = /data/logs/uwsgi/test.log
    plugins-dir = /data/sbin/uwsgi/plugins
    plugins = psgi

Here is how I am starting uWSGI:

    ./uwsgi --emperor /data/conf/uwsgi/ --binary-path /data/sbin/uwsgi/bin/uwsgi

Here is the log from starting uWSGI:

    *** Starting uWSGI 1.9.8 (64bit) on [Fri May  3 05:16:28 2013] ***
    compiled with version: 4.4.5 on 02 May 2013 03:37:28
    os: Linux-2.6.32-5-amd64 #1 SMP Mon Feb 25 00:26:11 UTC 2013
    nodename: dev
    machine: x86_64
    clock source: unix
    pcre jit disabled
    detected number of CPU cores: 2
    current working directory: /data/sbin/uwsgi/bin
    detected binary path: /data/sbin/uwsgi/bin/uwsgi
    *** WARNING: you are running uWSGI without its master process manager ***
    your memory page size is 4096 bytes
    detected max file descriptor number: 1024
    *** starting uWSGI Emperor ***
    *** has_emperor mode detected (fd: 6) ***
    [uWSGI] getting INI configuration from test.ini

Here is the log from the vassal:

    *** Starting uWSGI 1.9.8 (64bit) on [Fri May  3 05:16:28 2013] ***
    compiled with version: 4.4.5 on 02 May 2013 03:37:28
    os: Linux-2.6.32-5-amd64 #1 SMP Mon Feb 25 00:26:11 UTC 2013
    nodename: dev
    machine: x86_64
    clock source: unix
    pcre jit disabled
    detected number of CPU cores: 2
    current working directory: /data/conf/uwsgi
    detected binary path: /data/sbin/uwsgi/bin/uwsgi
    your memory page size is 4096 bytes
    detected max file descriptor number: 1024
    lock engine: pthread robust mutexes
    uwsgi socket 0 bound to TCP address :3001 fd 3
    uwsgi socket 1 bound to TCP address :4001 fd 4
    initialized Perl 5.16.3 main interpreter at 0x11c5570
    your server socket listen backlog is limited to 100 connections
    your mercy for graceful operations on workers is 60 seconds
    mapped 533888 bytes (521 KB) for 20 cores
    *** Operational MODE: threaded ***
    PSGI app 0 (index.psgi) loaded in 1 seconds at 0x14e0b00 (interpreter 0x11c5570)
    *** uWSGI is running in multiple interpreter mode ***
    spawned uWSGI master process (pid: 15175)
    spawned uWSGI worker 1 (pid: 15176, cores: 20)

Seems all really standard. The root path of this server should respond to my PSGI script. When I connect via HTTP, everything works as expected. When I connect via PSGI, however, I get this error:

    --- unable to find perl application ---
    [pid: 15176|app: -1|req: -1/2] 192.168.2.1 () {62 vars in 1087 bytes} [Fri May  3 05:18:17 2013] GET / => generated 21 bytes in 0 msecs (HTTP/1.1 500) 1 headers in 56 bytes (0 switches on core 3)

If I change the configuration from this:

    psgi=index.psgi

To this:

    mount = /=index.psgi

Then when I connect via HTTP, I get this error:

    --- unable to find perl application ---
    [pid: 15208|app: -1|req: -1/2] 192.168.2.1 () {36 vars in 648 bytes} [Fri May  3 05:19:55 2013] GET / => generated 21 bytes in 0 msecs (HTTP/1.1 500) 1 headers in 57 bytes (0 switches on core 3)

But when I connect via SCGI, now it works.

To summarize: when I use "index=psgi" it works with HTTP but not SCGI. When I use "mount=/=index.psgi" it works via SCGI but not HTTP.

**Comment by KES777** [comment]
In single APP environment all works fine

```
psgi = /home/user/public_html/my-app.psgi
manage-script-name = true
```

```
Mon Aug 24 13:05:19 2015 - *** Operational MODE: single process ***
2015/08/24 13:05:19 Application is started
Mon Aug 24 13:05:20 2015 - PSGI app 0 (/home/user/public_html/my-app.psgi) loaded in 1 seconds at 0x1b46f08
Mon Aug 24 13:05:20 2015 - spawned uWSGI master process (pid: 5810)
Mon Aug 24 13:05:20 2015 - spawned uWSGI worker 1 (pid: 5834, cores: 1)
```

But when trying to prepare to multiple apps

```
mount = ""=/home/user/public_html/my-app.psgi
manage-script-name = true
```

reloading server is OK

```
Mon Aug 24 13:08:26 2015 - *** Operational MODE: single process ***
Mon Aug 24 13:08:26 2015 - mounting /home/user/public_html/my-app.psgi on ""
2015/08/24 13:08:26 Application is started
Mon Aug 24 13:08:26 2015 - PSGI app 0 (/home/user/public_html/my-app.psgi) loaded in 0 seconds at 0x1e1d3d0 (inter
Mon Aug 24 13:08:26 2015 - spawned uWSGI master process (pid: 6002)
Mon Aug 24 13:08:26 2015 - spawned uWSGI worker 1 (pid: 6026, cores: 1)
```

but first request cause 

`Mon Aug 24 13:10:13 2015 - --- unable to find perl application ---`

Trying two different configurations:
```
mount = "app"=/home/user/public_html/my-app.psgi
manage-script-name = true
```

```
Mon Aug 24 13:12:04 2015 - *** Operational MODE: single process ***
Mon Aug 24 13:12:04 2015 - mounting /home/user/public_html/my-app.psgi on "app"
2015/08/24 13:12:04 Application is started
Mon Aug 24 13:12:04 2015 - PSGI app 0 (/home/user/public_html/my-app.psgi) loaded in 0 seconds at 0x2e2d330 (inter
Mon Aug 24 13:12:04 2015 - spawned uWSGI master process (pid: 6193)
Mon Aug 24 13:12:04 2015 - spawned uWSGI worker 1 (pid: 6217, cores: 1)
Mon Aug 24 13:12:30 2015 - --- unable to find perl application ---
```

And second:
```
mount = "/"=/home/user/public_html/my-app.psgi
manage-script-name = true

```

```
Mon Aug 24 13:15:15 2015 - *** Operational MODE: single process ***
Mon Aug 24 13:15:15 2015 - mounting /home/user/public_html/my-app.psgi on "/"
2015/08/24 13:15:15 Application is started
Mon Aug 24 13:15:15 2015 - PSGI app 0 (/home/user/public_html/my-app.psgi) loaded in 0 seconds at 0x25a2d60 (inter
Mon Aug 24 13:15:15 2015 - spawned uWSGI master process (pid: 6369)
Mon Aug 24 13:15:15 2015 - spawned uWSGI worker 1 (pid: 6393, cores: 1)
```

Gives same result:

`--- unable to find perl application ---`

**Comment by unbit** [comment]
Can you remove double quotes ? They will be included in the mountpoint

**Comment by KES777** [comment]
```
mount = /=/home/user/public_html/my-app.psgi
manage-script-name = true
```

```
Mon Aug 24 13:35:38 2015 - *** Operational MODE: single process ***
Mon Aug 24 13:35:38 2015 - mounting /home/user/public_html/my-app.psgi on /
2015/08/24 13:35:39 Application is started
Mon Aug 24 13:35:39 2015 - PSGI app 0 (/home/user/public_html/my-app.psgi) loaded in 1 seconds at 0x26cee08 (inter
Mon Aug 24 13:35:39 2015 - spawned uWSGI master process (pid: 6783)
Mon Aug 24 13:35:39 2015 - spawned uWSGI worker 1 (pid: 6807, cores: 1)
```
cause wrong:
```
SCRIPT_NAME=/
PATH_INFO=test.html
```

And ugly config make all FINE!
```
mount = =/home/user/public_html/my-app.psgi
manage-script-name = true
```

Notice UGLY config also
`Mon Aug 24 13:37:21 2015 - mounting /home/user/public_html/my-app.psgi on`

```
Mon Aug 24 13:37:21 2015 - *** Operational MODE: single process ***
Mon Aug 24 13:37:21 2015 - mounting /home/user/public_html/my-app.psgi on  <<< NOTING IS HERE
2015/08/24 13:37:21 Application is started
Mon Aug 24 13:37:21 2015 - PSGI app 0 (/home/user/public_html/my-app.psgi) loaded in 0 seconds at
Mon Aug 24 13:37:21 2015 - spawned uWSGI master process (pid: 6964)
Mon Aug 24 13:37:21 2015 - spawned uWSGI worker 1 (pid: 6988, cores: 1)
```

```
SCRIPT_NAME=
PATH_INFO=/test.html
```

So, 
`psgi = /home/feelsafe/public_html/feel-safe.net.psgi`
is equivalent in `mount` manner:
`mount = =/home/feelsafe/public_html/feel-safe.net.psgi`

**Thank you**

PS. Can you route interface to a common denominator?
remove "psgi/wsgi-file/rack" and left only one "mount"
But it looks like all these are shortcuts to [route] (http://uwsgi-docs.readthedocs.org/en/latest/InternalRouting.html#uwsgi) or (seems similar too) route-run = uwsgi:</path/to/script.psgi>??

**Comment by unbit** [comment]
So if i followed all well, the "/" mountpoint is not honoured as empty SCRIPT_NAME, right ?

**Comment by KES777** [comment]
right

**Comment by unbit** [comment]
Ok, i was able to reproduce it, i'll push a fix in the next few minutes. Thanks again
